### DISCUSSION TITLE
mergeQueryHs merges isotopes

### DISCUSSION THREAD

**Discussion by jones-gareth** [rdkit/rdkit#5554] [General]
`MergeQueryHs` will replace deuterium (and other hydrogen isotopes) by a hydrogen count constraint on the heavy neighbor, losing the isotope.

I like `mergeQueryHs` as it allows an explicit hydrogen in a query to match either explicit or implicit hydrogens in a target compound.  However, I typically don't want to merge isotopes.  If you have deuterium in a query, then you only want to mach deuterium in a target compound.

I think it would be a good idea to change `MolOps::mergeQueryHs` to have a`mergeIsotopes` argument so that you can skip merging isotopes.   I'm happy to make that change.

**Comment by greglandrum** [comment]
@jones-gareth i agree that this would be an improvement. I believe it should also be the default behavior

**Reply by jones-gareth** [reply]
Shall I go ahead and implement, or would you prefer to do it?

One thing that concerns me about making it the default behavior is that I've seen the isotope used as a mapping in substructure search.  I guess that would come out in tests.

**Reply by greglandrum** [reply]
Either way... I can do it this week, but I'm always happy for PRs from other people too. :-)

**Reply by jones-gareth** [reply]
I'll do a PR- I have to do some C++ programming soon and this will give me a push.
